presto-main/src/main/java/com/facebook/presto/operator/SimplePagesIndexComparator.java

```diff
@@ -30,45 +32,47 @@
 public class SimplePagesIndexComparator
         implements PagesIndexComparator
 {
-    private final List<Integer> sortChannels;
-    private final List<SortOrder> sortOrders;
-    private final List<Type> sortTypes;
+    private final int[] sortChannels;
+    private final SortOrder[] sortOrders;
+    private final Type[] sortTypes;
 
     public SimplePagesIndexComparator(List<Type> sortTypes, List<Integer> sortChannels, List<SortOrder> sortOrders)
     {
-        this.sortTypes = ImmutableList.copyOf(requireNonNull(sortTypes, "sortTypes is null"));
-        this.sortChannels = ImmutableList.copyOf(requireNonNull(sortChannels, "sortChannels is null"));
-        this.sortOrders = ImmutableList.copyOf(requireNonNull(sortOrders, "sortOrders is null"));
+        this.sortChannels = Ints.toArray(requireNonNull(sortChannels, "sortChannels is null"));
+        this.sortOrders = requireNonNull(sortOrders, "sortOrders is null").toArray(new SortOrder[0]);
+        this.sortTypes = requireNonNull(sortTypes, "sortTypes is null").toArray(new Type[0]);
     }
 
     @Override
     public int compareTo(PagesIndex pagesIndex, int leftPosition, int rightPosition)
     {
-        long leftPageAddress = pagesIndex.getValueAddresses().get(leftPosition);
+        AdaptiveLongBigArray valueAddresses = pagesIndex.getValueAddresses();
+        long leftPageAddress = valueAddresses.get(leftPosition);
+        long rightPageAddress = valueAddresses.get(rightPosition);
+
         int leftBlockIndex = decodeSliceIndex(leftPageAddress);
         int leftBlockPosition = decodePosition(leftPageAddress);
 
-        long rightPageAddress = pagesIndex.getValueAddresses().get(rightPosition);
         int rightBlockIndex = decodeSliceIndex(rightPageAddress);
         int rightBlockPosition = decodePosition(rightPageAddress);
 
-        for (int i = 0; i < sortChannels.size(); i++) {
-            int sortChannel = sortChannels.get(i);
-            Block leftBlock = pagesIndex.getChannel(sortChannel).get(leftBlockIndex);
-            Block rightBlock = pagesIndex.getChannel(sortChannel).get(rightBlockIndex);
+        for (int i = 0; i < sortChannels.length; i++) {
+            int sortChannel = sortChannels[i];
+            SortOrder sortOrder = sortOrders[i];
+            Type sortType = sortTypes[i];
+            ObjectArrayList<Block> indexChannel = pagesIndex.getChannel(sortChannel);
```

Comment:
weird the interface returns `ObjectArrayList`. Given it's the only place using this interface, maybe change the interface to return `List<...>`

Reply:
Sure, changed to `List<Block>`